provider/bigquery.go

```diff
@@ -903,48 +913,48 @@ func (store *bqOfflineStore) GetTransformationTable(id ResourceID) (Transformati
 func (store *bqOfflineStore) UpdateTransformation(config TransformationConfig) error {
 	name, err := store.createTransformationName(config.TargetTableID)
 	if err != nil {
-		return err
+		return fferr.NewTransformationNotFoundError(config.TargetTableID.Name, config.TargetTableID.Variant, nil)
 	}
 	err = store.query.transformationUpdate(store.client, name, config.Query)
 	if err != nil {
-		return err
+		return fferr.NewTransformationNotFoundError(config.TargetTableID.Name, config.TargetTableID.Variant, nil)
 	}
 
 	return nil
 }
 
 func (store *bqOfflineStore) CreatePrimaryTable(id ResourceID, schema TableSchema) (PrimaryTable, error) {
 	if err := id.check(Primary); err != nil {
-		return nil, err
+		return nil, fferr.NewDatasetNotFoundError(id.Name, id.Variant, err)
 	}
 	if exists, err := store.tableExists(id); err != nil {
-		return nil, err
+		return nil, fferr.NewDatasetNotFoundError(id.Name, id.Variant, err)
 	} else if exists {
 		return nil, &TableAlreadyExists{id.Name, id.Variant}
 	}
 	if len(schema.Columns) == 0 {
-		return nil, fmt.Errorf("cannot create primary table without columns")
+		return nil, fferr.NewDatasetNotFoundError(id.Name, id.Variant, fmt.Errorf("cannot create primary table without columns"))
 	}
 	tableName, err := GetPrimaryTableName(id)
 	if err != nil {
-		return nil, err
+		return nil, fferr.NewDatasetNotFoundError(id.Name, id.Variant, err)
 	}
 	table, err := store.newBigQueryPrimaryTable(store.client, tableName, schema)
 	if err != nil {
-		return nil, err
+		return nil, fferr.NewExecutionError(string(store.Type()), id.Name, id.Variant, id.Type.String(), err)
```

Comment:
These errors should probably be inside these store functions and just returned here